src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
+        private const string LinuxName = "linux";
+        private const string OSXName = "osx";
+
+        private static readonly OSName s_windows = new OSName(WindowsName);
+        private static readonly OSName s_linux = new OSName(LinuxName);
+        private static readonly OSName s_osx = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name.Equals("unix", StringComparison.OrdinalIgnoreCase))
+                _name = LinuxName;
+            else if (name.Equals("mac", StringComparison.OrdinalIgnoreCase) ||
+                name.Equals("macosx", StringComparison.OrdinalIgnoreCase))
```

Comment:
+1 I think it seems a unnecessary to actually match names here and in fact may cause issues in the future if we design to not map mac to macosx or unix to linux. Instead we should leave that all up to IsOSPlatform as @ellismg suggested.